nats.go

```diff
@@ -3644,6 +3661,9 @@ func (s *Subscription) NextMsg(timeout time.Duration) (*Msg, error) {
 		if err := s.processNextMsgDelivered(msg); err != nil {
 			return nil, err
 		} else {
+			if jsi != nil {
+				return processControlFlow(msg)
```

Comment:
Should we loop back and see if real message is behind this one instead of returning here?

Reply:
Added so that we peek at the channel and return the next queued message if the channel is ready to send.